### ISSUE TITLE
Create compute instance on custom subnet?

### ISSUE THREAD

**Issue by Vanders** [fog/fog-google#117]
Ruby 2.3.0
fog-google 0.2.0

I can't seem to find any way to create a new compute instance with the network configured to use a custom subnet. I have a network 'rnd1' with multiple sub-networks in it; if I pass `network: 'rnd1'` to Fog::Compute.servers.create I get:

`Invalid value for field 'resource.networkInterfaces[0]': ''. Subnetwork should be specified for custom subnetmode network`

I assume I need to also pass the name of the subnetwork, but looking over the code I see no way to do this.

**Comment by icco** [comment]
Hmm, yes, this is a bug. When we wrote this, it wasn't possible to specify the subnet. Do you have an example snippet that you're calling to get this error for the record? Where in the process would you expect to be able to specify this?

**Comment by Vanders** [comment]
So if we assume I have a custom network called 'rnd1' with the subnets as follows:
```
rnd1:
  edge
  operations
  builders
```
and I want to create an instance in the subnetwork 'builders', then I'd expect to be able to do something like:
```
fog = Fog::Compute.new(provider: 'Google')
...
instance = fog.servers.create(name: name,
                                                disks: [disk],
                                                machine_type: 'f1-micro',
                                                zone_name: 'us-central1-a',
                                                network: 'rnd1',
                                                subnet: 'builders',
                                                metadata: metadata,
                                                tags: ['build','build-host'])
```
I.e. simply pass the name of subnetwork in a 'subnet' argument, alongside the name of the custom network in the 'network' argument.

**Comment by icco** [comment]
Interesting note is that fog-aws has subnet as a first class model. Also google is considering network to be a first class model as well. Might be time to break those out better.

 - https://github.com/fog/fog-aws/blob/master/lib/fog/aws/compute.rb
 - https://github.com/fog/fog-aws/blob/master/lib/fog/aws/models/compute/subnet.rb

cc @Temikus

**Comment by Leila-Kuntar** [comment]
Is it still impossible to specify a subnetwork? I have this issue with [vagrant-google plugin](https://github.com/mitchellh/vagrant-google/issues/164).

**Comment by icco** [comment]
vagrant-google [is on 0.4.2](https://github.com/mitchellh/vagrant-google/blob/master/vagrant-google.gemspec#L31) which doesn't have support for subnetworks. Since then though, it does not look like anyone has added support for them either: https://github.com/fog/fog-google/compare/d5defcf67f9b93ba1f2dcf49d43d28e50f01d415...HEAD

We're a completely volunteer run operation though, so the best chance of getting this fixed is through pull requests.

**Comment by Wyosotis** [comment]
@icco I hope release comes earlier than winter=)

**Comment by icco** [comment]
Sure, I can put out a release @Wyosotis.

**Comment by icco** [comment]
Released as 0.5.3
